Patterns/Spanish/Spanish-DateTime.yaml

```diff
@@ -8,7 +8,9 @@ DayRegex: !simpleRegex
 MonthNumRegex: !simpleRegex
   def: (?<month>01|02|03|04|05|06|07|08|09|10|11|12|1|2|3|4|5|6|7|8|9)\b
 YearRegex: !simpleRegex
-  def: (?<year>((1[5-9]|20)\d{2})|[0-27-9]\d)
+  def: \b(?<year>(1[5-9]|20)\d{2})(?!\.0\b)\b
```

Comment:
Can you add a comment in the YAML file here to explain (?!\.0\b)?
Also, this fix also should propagate across languages.